filters/f_hwtransfer.c

```diff
@@ -490,24 +502,16 @@ static bool probe_formats(struct mp_filter *f, int hw_imgfmt, bool use_conversio
                                 AV_HWFRAME_TRANSFER_DIRECTION_TO, &fmts, 0) >= 0 &&
                                 fmts[0] != AV_PIX_FMT_NONE)
             {
-                // Don't add the format to p->fmts if we know it cannot be used.
-                if (ctx->supported_hwupload_formats) {
-                    for (int i = 0; ctx->supported_hwupload_formats[i]; i++) {
-                        if (ctx->supported_hwupload_formats[i] == imgfmt)
-                            MP_TARRAY_APPEND(p, p->fmts, p->num_fmts, imgfmt);
-                    }
-                } else {
-                    MP_TARRAY_APPEND(p, p->fmts, p->num_fmts, imgfmt);
-                }
-
-                int index = hw_transfer_index;
-                hw_transfer_index++;
-
+                int index = p->num_fmts;
+                MP_TARRAY_APPEND(p, p->fmts, p->num_fmts, imgfmt);
```

Comment:
This won't work for me. The not supported format can't be in `p->fmts`. The upload fails on my end.

Reply:
Could you show log with `hwupload=trace`? I'm not sure I understand which part of upload has to be excluded on your end.

Reply:
[Sure](https://github.com/user-attachments/files/17232387/out.log)

The earlier version of this PR worked. What has to be filtered out is `p->fmts` because it will attempt to use anything in there for hwupload.

Reply:
Okay yeah it works now.